server/src/main/java/org/elasticsearch/search/aggregations/metrics/WrapperTDigestArrays.java

```diff
@@ -0,0 +1,365 @@
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+package org.elasticsearch.search.aggregations.metrics;
+
+import org.apache.lucene.util.Accountable;
+import org.apache.lucene.util.RamUsageEstimator;
+import org.elasticsearch.common.breaker.CircuitBreaker;
+import org.elasticsearch.common.breaker.CircuitBreakingException;
+import org.elasticsearch.common.breaker.NoopCircuitBreaker;
+import org.elasticsearch.core.Releasable;
+import org.elasticsearch.core.Releasables;
+import org.elasticsearch.tdigest.arrays.TDigestArrays;
+import org.elasticsearch.tdigest.arrays.TDigestByteArray;
+import org.elasticsearch.tdigest.arrays.TDigestDoubleArray;
+import org.elasticsearch.tdigest.arrays.TDigestIntArray;
+import org.elasticsearch.tdigest.arrays.TDigestLongArray;
+
+import java.util.Arrays;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+/**
+ * TDigestArrays with raw arrays and circuit breaking.
+ */
+public class WrapperTDigestArrays implements TDigestArrays {
+
+    /**
+     * Default no-op CB instance of the wrapper.
+     *
+     * @deprecated This instance shouldn't be used, and will be removed after all usages are replaced.
+     */
+    @Deprecated
+    public static final WrapperTDigestArrays INSTANCE = new WrapperTDigestArrays(new NoopCircuitBreaker("default-wrapper-tdigest-arrays"));
+
+    private final CircuitBreaker breaker;
+
+    public WrapperTDigestArrays(CircuitBreaker breaker) {
+        this.breaker = breaker;
+    }
+
+    @Override
+    public WrapperTDigestDoubleArray newDoubleArray(int initialCapacity) {
+        return validate(new WrapperTDigestDoubleArray(breaker, initialCapacity));
+    }
+
+    @Override
+    public WrapperTDigestIntArray newIntArray(int initialSize) {
+        return validate(new WrapperTDigestIntArray(breaker, initialSize));
+    }
+
+    @Override
+    public TDigestLongArray newLongArray(int initialSize) {
+        return validate(new WrapperTDigestLongArray(breaker, initialSize));
+    }
+
+    @Override
+    public TDigestByteArray newByteArray(int initialSize) {
+        return validate(new WrapperTDigestByteArray(breaker, initialSize));
+    }
+
+    public WrapperTDigestDoubleArray newDoubleArray(double[] array) {
+        return validate(new WrapperTDigestDoubleArray(breaker, array));
+    }
+
+    public WrapperTDigestIntArray newIntArray(int[] array) {
+        return validate(new WrapperTDigestIntArray(breaker, array));
+    }
+
+    /**
+     * Safely adjusts the breaker with an array.
+     * <p>
+     *     Copied from {@link org.elasticsearch.common.util.BigArrays}.
+     * </p>
+     */
+    private <T extends Accountable & Releasable> T validate(T array) {
+        boolean success = false;
+        try {
+            adjustBreaker(array.ramBytesUsed());
+            success = true;
+        } finally {
+            if (success == false) {
+                Releasables.closeExpectNoException(array);
+            }
+        }
+        return array;
+    }
```

Comment:
Commented with @nik9000:
"Account after creation" is the current BigArrays _order_ for wrapper arrays, so I just cloned it here.
However, it would be better to increment the CB _before_ creation. I'll check it later, and potentially change it in this PR

Reply:
++